MyProject/cls/Restaurant.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace cls
+{
+       public class Restaurant
+    {
+        public string Name;
```

Comment:
as you have methods to set value of each of this fields, i guess what you need here is a properties with private set.